wildvvad/sample.py

```diff
@@ -0,0 +1,268 @@
+import collections
+import os
+import glob
+import pickle
+
+import cv2
+import numpy as np
+import face_alignment
+from matplotlib import pyplot as plt
+from utils import utils
+import vg
+
+import open3d as o3d
+
+
+class Sample:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def load_sample_objects_from_disk(folder_path: str):
+        """
+        Loads all sample objects (pickle files) from a folder
+
+        Args:
+            folder_path (str): path to sample file
+        Returns:
+            video_samples_objects (): Generator with sample objects
+        """
+
+        old_cwd = os.getcwd()
+        os.chdir(folder_path)
+        for file in glob.glob("*.pickle"):
+            with open(file, 'rb') as pickle_file:
+                yield pickle.load(pickle_file)
+
+        # Reset to previous working directory
+        os.chdir(old_cwd)
+
+    @staticmethod
+    def load_video_sample_from_disk(file_path: str):
+        """
+        Loads all video samples from a specified folder.
+
+        Args:
+            file_path (str): path to sample file
+        Returns:
+            video_samples (): Generator with frames of one sample
+        """
+
+        # with os.scandir(path) as folder:
+        # for file in path:  # folder:
+        count = 0
+        # ToDo check for file type
+        # if file.name.endswith(".XXX"):
+        video_path = os.path.join(file_path)
+        vid_obj = cv2.VideoCapture(video_path)
+
+        if not vid_obj.isOpened():
+            print("could not open :", video_path)
+            return
+
+        frame_num = int(vid_obj.get(cv2.CAP_PROP_FRAME_COUNT))
+        vid_fps = vid_obj.get(cv2.CAP_PROP_FPS)
+
+        # label = True
+        # config = {"fps": vid_fps}
+        print("FPS are ", vid_fps)
+
+        success = vid_obj.grab()
+
+        if not success:
+            raise Exception(
+                "Couldn't grab frame of file {}".format(video_path))
+
+        # grab frames from start to end frame
+        while success:
+            _, image = vid_obj.retrieve()
+
+            # ToDo needed?
+            if count <= frame_num:
+                pass
+                # data.append(image)
+            count += 1
+            if count > frame_num:
+                break
+
+            success = vid_obj.grab()
+
+            yield image
+
+    @staticmethod
+    def get_face_landmark_from_sample(image):
+        fa = face_alignment.FaceAlignment(face_alignment.LandmarksType.THREE_D,
+                                          flip_input=False, device='cpu')
+
+        return fa.get_landmarks(image)
+
+    def visualize_3d_landmarks(self, image, landmarks, landmarks_test):
```

Comment:
missing doc_string and this could be a static method as well if it only uses another static method